core/src/main/java/ai/timefold/solver/core/impl/domain/common/accessor/ReflectionBeanPropertyMemberAccessor.java

```diff
@@ -34,33 +36,38 @@ public ReflectionBeanPropertyMemberAccessor(Method getterMethod, boolean getterO
         } catch (IllegalAccessException e) {
             throw new IllegalStateException("""
                     Impossible state: method (%s) not accessible.
-                    %s
-                    """
-                    .strip()
+                    %s"""
                     .formatted(getterMethod, MemberAccessorFactory.CLASSLOADER_NUDGE_MESSAGE), e);
         }
         Class<?> declaringClass = getterMethod.getDeclaringClass();
         if (!ReflectionHelper.isGetterMethod(getterMethod)) {
-            throw new IllegalArgumentException("The getterMethod (" + getterMethod + ") is not a valid getter.");
+            throw new IllegalArgumentException("The getterMethod (%s) is not a valid getter."
+                    .formatted(getterMethod));
         }
         propertyType = getterMethod.getReturnType();
         propertyName = ReflectionHelper.getGetterPropertyName(getterMethod);
         if (getterOnly) {
             setterMethod = null;
             setterMethodHandle = null;
         } else {
-            setterMethod = ReflectionHelper.getSetterMethod(declaringClass, getterMethod.getReturnType(), propertyName);
+            setterMethod = ReflectionHelper.getDeclaredSetterMethod(declaringClass, getterMethod.getReturnType(), propertyName);
             if (setterMethod != null) {
+                var getterAccess = AccessModifier.forMethod(getterMethod);
+                var setterAccess = AccessModifier.forMethod(setterMethod);
+                if (getterAccess != setterAccess) {
+                    throw new IllegalArgumentException(
+                            "The getter method (%s) has access modifier (%s) which does not match the setter method's (%s) access modifier (%s) on class (%s)."
+                                    .formatted(getterMethod.getName(), getterAccess, setterMethod.getName(), setterAccess,
+                                            getterMethod.getDeclaringClass().getSimpleName()));
+                }
                 try {
                     setterMethod.setAccessible(true); // Performance hack by avoiding security checks
                     this.setterMethodHandle = lookup.unreflect(setterMethod)
                             .asFixedArity();
                 } catch (IllegalAccessException e) {
                     throw new IllegalStateException("""
                             Impossible state: method (%s) not accessible.
-                            %s
-                            """
-                            .strip()
+                            %s"""
                             .formatted(setterMethod, MemberAccessorFactory.CLASSLOADER_NUDGE_MESSAGE), e);
                 }
             } else {
```

Comment:
At this point, there is no setter method, but we do not fail fast. Is that correct?